packages/smart-contracts/test/contracts/ERC20BatchConversionPayments.test.ts

```diff
@@ -0,0 +1,611 @@
+import { ethers, network } from 'hardhat';
+import {
+  ERC20FeeProxy__factory,
+  Erc20ConversionProxy__factory,
+  EthConversionProxy__factory,
+  EthereumFeeProxy__factory,
+  ERC20FeeProxy,
+  EthereumFeeProxy,
+  ChainlinkConversionPath,
+  TestERC20,
+  Erc20ConversionProxy,
+  EthConversionProxy,
+  TestERC20__factory,
+  BatchConversionPayments,
+} from '../../src/types';
+import { BigNumber, BigNumberish, BytesLike, ContractTransaction, Signer } from 'ethers';
+import { expect } from 'chai';
+import { CurrencyManager } from '@requestnetwork/currency';
+import { chainlinkConversionPath, batchConversionPaymentsArtifact } from '../../src/lib';
+import { localERC20AlphaArtifact, secondLocalERC20AlphaArtifact } from './localArtifacts';
+import Utils from '@requestnetwork/utils';
+
+// set to true to log batch payments's gas consumption
+const logGas = false;
+
+describe('contract: BatchConversionPayments', () => {
+  let from: string;
+  let to: string;
+  let feeAddress: string;
+  let batchAddress: string;
+  let signer: Signer;
+  let xSigner: Signer;
+
+  // variables used to set up batch conversion proxy, and also requests payment
+  const batchFee = 50;
+  const batchConvFee = 100;
+  const amountInFiat = '100000000'; // 1 with 8 decimal
+  const feesAmountInFiat = '100000'; // 0.001 with 8 decimal
+  const thousandWith18Decimal = '1000000000000000000000';
+  const referenceExample = '0xaaaa';
+
+  // variables to set up proxies and paths
+  const currencyManager = CurrencyManager.getDefault();
+
+  const ETH_hash = currencyManager.fromSymbol('ETH')!.hash;
+  const USD_hash = currencyManager.fromSymbol('USD')!.hash;
+  const EUR_hash = currencyManager.fromSymbol('EUR')!.hash;
+  let DAI_address: string;
+  let FAU_address: string;
+
+  let testErc20ConversionProxy: Erc20ConversionProxy;
+  let testEthConversionProxy: EthConversionProxy;
+  let testBatchConversionProxy: BatchConversionPayments;
+  let testERC20: TestERC20;
+  let testERC20b: TestERC20;
+  let erc20FeeProxy: ERC20FeeProxy;
+  let ethereumFeeProxy: EthereumFeeProxy;
+  let chainlinkPath: ChainlinkConversionPath;
+
+  // variables used to check testERC20 balances
+  let fromOldBalance: BigNumber;
+  let toOldBalance: BigNumber;
+  let feeOldBalance: BigNumber;
+
+  let fromDiffBalanceExpected: BigNumber;
+  let toDiffBalanceExpected: BigNumber;
+  let feeDiffBalanceExpected: BigNumber;
+
+  // variables needed for chainlink and
+  let path: string[];
+  type ConvToPay = [BigNumber, BigNumber] & {
+    result: BigNumber;
+    oldestRateTimestamp: BigNumber;
+  };
```

Comment:
I don't understand this type and it looks suspicious that we need a type definition for tests. Wouldn't it be enough with a fixed rate for example? It's not like we have to really test the conversion in these tests.